data-protocols/ids/ids-api-multipart-endpoint-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/handler/ContractRequestHandler.java

```diff
@@ -145,7 +150,7 @@ public boolean canHandle(@NotNull MultipartRequest multipartRequest) {
         negotiationManager.requested(verificationResult.getContent(), requestObj);
 
         return MultipartResponse.Builder.newInstance()
-                .header(ResponseMessageUtil.createRequestInProcessMessage(connectorId, message))
+                .header(responseMessageFactory.createRequestInProcessMessage(message))
```

Comment:
same as ContractOfferHandler

Reply:
Thanks for pointing this out. I was so focuses on updating the message content, that I forgot to handle the new exceptions in the existing classes accordingly.

The exception should now be added. I also change the type of the `InvalidCorrelationMessageException` to `IllegalArgumentException`, so that its mapped to a 400 status code in the  `EdcApiExceptionMapper`. I assume this will become necessary when the code in a catch-clause is not able to create a rejection message, too.